File: submissions/x0r/problem1.cpp
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;
import static java.lang.System.out;

class program1 {
	private static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	public static void main(String[] args) throws Exception {
		int t = Integer.parseInt(br.readLine());
		while (t-- > 0) {
			String s = br.readLine();
			int o = s.indexOf('$')+1;
			String tmp = "";
			for (; o < s.length(); ++o) {
				char ch = s.charAt(o);
				if (Character.isDigit(ch) || Character.isWhitespace(ch))
					tmp += ch;
				else break;
			}
			String f = "";
			for (char ch : tmp.toCharArray()) {
				if (Character.isDigit(ch)) {
					f += ch;
				}
			}
			o = 0;
			while (o < f.length() && f.charAt(o) == 48) o++;
			if (o == f.length()) {
				System.out.println("$0");
			} else {
				System.out.println("$" + f.substring(o));
			}
		}
	}
}

File: submissions/ankit biswas/problem1.java
import java.io.*;
class program1
{BufferedReader br=new BufferedReader(new InputStreamReader(System.in)); 
 int n=0;int si=0;int j=0;int Si=0;int c=0;
 void main()throws IOException
 {System.out.println("Give the number of inputs ");
  n=Integer.parseInt(br.readLine()); String alpha[]=new String[n];  
  for(int i=0;i<n;i++)
    {System.out.println("enter the line "+(i+1));
     alpha[i]=br.readLine();
    }
  for(int i=0;i<n;i++)
     {for(j=1;;j++)
        { if(alpha[i].charAt(j-1)=='\n')
             break;
            }
       
   
    {for(c=0;c<j-1;c++)
        {if(alpha[i].charAt(c)=='$')
            {while((int)(alpha[i].charAt(c))>48&&(int)(alpha[i].charAt(c))<=57)
                   {Si=Si*10+(int)(alpha[i].charAt(c));c++;
                       if(c==j)
                         {System.out.println('$'+Si);
                         break;}
                    }
                }
            }
            c=0;}}}}
            
               
                   
                   
     